### PR TITLE
diagnostics: add support for running diagnostics on remote machine

### PR DESCRIPTION
**Pull request by assafgi**

### CODE REVIEW

File: talker_agent/talker.py [hunk 1]
```diff
@@ -731,6 +731,16 @@ def start_job(self, job_data_raw):
...
             job = RebootJob(agent=self, **job_data)
             job.start()
+        elif cmd == 'diagnostics':
+            logger.info('Got diagnostics request')
+            try:
+                script_path = config.parser.get('diagnostics', 'script_path')
+            except NoSectionError:
+                message = 'diagnostics script path is missing from config file'
+                logger.exception(message)
+                raise Exception(message)
```
[the remark this answers is not in the document]
Comment (assafgi) [reply talker_agent/talker.py]: yes, but if I remove the diagnostics it is not relevant, this `elif` will be deleted..

File: talker_agent/talker.py [hunk 2]
```diff
@@ -870,6 +870,21 @@ def unregister_fileno(self, fileno):
...
 
 
+def run_diagnostics():
+    try:
+        script_path = config.parser.get('diagnostics', 'script_path')
+    except NoSectionError:
+        logger.warning('diagnostics script path is missing from config file')
+        return
+
+    if os.path.isfile(script_path):
```
Comment (DoronCohen) [inline talker_agent/talker.py]: We can log a warning if the file doesnt exist

File: talker_agent/talker.py [hunk 3]
```diff
@@ -870,6 +870,21 @@ def unregister_fileno(self, fileno):
             pass  # closed while processing data
 
 
+def run_diagnostics():
+    try:
+        script_path = config.parser.get('diagnostics', 'script_path')
+    except NoSectionError:
+        logger.warning('diagnostics script path is missing from config file')
```
Comment (DoronCohen) [inline talker_agent/talker.py]: The configuration is optional so if it isnt set we shouldnt log anything

File: talker_agent/talker.py [hunk 4]
```diff
@@ -870,6 +870,21 @@ def unregister_fileno(self, fileno):
...
+        return
+
+    if os.path.isfile(script_path):
+        logger.info("Running diagnostics")
+        try:
+            subprocess.Popen(["bash", "-ce", script_path], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```
Comment (DoronCohen) [inline talker_agent/talker.py]: I am not sure the process should use Talker's `stdout` and `stderr`. Especially when we know we are about to die.

File: talker_agent/talker.py [hunk 5]
```diff
@@ -870,6 +870,22 @@ def unregister_fileno(self, fileno):
...
+        script_path = config.parser.get('diagnostics', 'script_path')
+    except NoSectionError:
+        return
+
+    if os.path.isfile(script_path):
+        logger.info("Running diagnostics")
+        try:
+            subprocess.Popen(["bash", "-ce", script_path])
+        except:
+            pass
```
Comment (DoronCohen) [inline talker_agent/talker.py]: I would log the exception message here

**Pull request merged**